### PR TITLE
Merge zn_pub_thr into z_put_thr

### PR DESCRIPTION
**Pull request by william-wyliang**
This PR merges throughput/zn_pub_thr into throughput/z_put_thr
- New options
  --use-expr: use declare_expr() for the writer's key
  --declare-publication: call declare_publication() before writing
- zn_pub_thr is removed

Verification Items
- [x] Pass compilation with `cargo build --bin z_put_thr`
- [x] Verified by the following command with different option combinations
  `$ ./target/debug/z_sub_thr -s my-test -m peer -l tcp/localhost:7447 -p 8 -n now [--use-expr] [--no-callback]`
  `$ ./target/debug/z_put_thr -m peer -l tcp/localhost:7447 -p 8 [--use-expr] [--declare-publication]`

### CODE REVIEW

**Review by eddie9712 — APPROVED** [review]
LGTM

File: throughput/src/bin/z_put_thr.rs [hunk 1]
```diff
@@ -84,6 +95,18 @@ async fn main() {
...
+            session.declare_publication(KEY_EXPR);
+        }
+        session.put(KEY_EXPR, value.clone())
+    } else {
+        let expr_id = session.declare_expr(KEY_EXPR).await.unwrap();
+        if declare_publication {
+            session.declare_publication(expr_id);
+        }
+        session.put(expr_id, value.clone())
+    };
```
Comment (YuanYuYuan) [inline throughput/src/bin/z_put_thr.rs]: I'm afraid that you use the opposite in this if/else branch. 😅

**Review by YuanYuYuan — CHANGES_REQUESTED** [review]
@william-wyliang  Is it better to change the name from z_put_thr to z_pub_thr? So that the remaining codes are consistently in pub & sub.

File: throughput/src/bin/z_put_thr.rs (hunk 1 above)
Comment (william-wyliang) [reply throughput/src/bin/z_put_thr.rs]: You are right. Will fix this.

Comment (william-wyliang) [reply throughput/src/bin/z_put_thr.rs]: Regarding to the name pub or put, according to the architecture concept and the API design of zenoh, the data-centric view favors put than pub. As a result, I would suggest we keep it as z_put. Or, maybe we can postpone this for later discussion?

Comment (YuanYuYuan) [reply throughput/src/bin/z_put_thr.rs]: okay

**Pull request merged**